readthedocs/projects/views/private.py

```diff
@@ -737,8 +737,12 @@ class DomainUpdate(DomainMixin, UpdateView):
     pass
 
 
-class DomainDelete(DomainMixin, DeleteView):
-    pass
+class DomainDelete(DomainMixin, View):
+    
+    def get(self, request, *args, **kwargs):
+        domain = get_object_or_404(Domain, pk=kwargs.get('domain_pk'))
+        domain.delete()
+        return HttpResponseRedirect(self.get_success_url())
```

Comment:
Isn't this done automatically by the `DeleteView`?

Reply:
Because of the `DeleteView`, it first ask for confirmation (`Do you want to delete this domain`) and then deletes it on confirmation.

But it was required to remove the confirmation (issue description), and so I implement the `get()` method. I tried implementing the `POST` request with the help of `<form>` but due to the CSS, both of the buttons looks completely different. So, I dropped the idea.
Although, I still feels that a `GET` request is not the way to do it.

Would like any other way of implementing, if possible.

Reply:
Can you transform this explanation into a comment in the code?

Some time ago I did the same and wrote a small comment at https://github.com/rtfd/readthedocs.org/blob/86bfdffaee2a6a5143f6c7c75af2600817101ce1/readthedocs/projects/views/private.py#L940

(you can write a comment that is better than mine ;) 

I copied that line from another methods, like https://github.com/rtfd/readthedocs.org/blob/86bfdffaee2a6a5143f6c7c75af2600817101ce1/readthedocs/projects/views/private.py#L852

So, maybe you can follow the same pattern.

Reply:
BTW, we don't want to delete an object from the database on a GET request.

Reply:
> BTW, we don't want to delete an object from the database on a GET request.

Javascript to the rescue. I did managed to use the POST request without altering the CSS.
Now the workflow is like this:
1. Click on 'Remove'
2. `alert` asking for confirmation.
3. If Yes, deletes the domain with a POST request. 

Also, I have added the comment.